aws/resource_aws_kinesis_analyticsv2_application_test.go

```diff
@@ -0,0 +1,1983 @@
+  runtime                = "SQL-1_0"
+  service_execution_role = "${aws_iam_role.kinesis_analyticsv2_application.arn}"
+
+  application_configuration {
+    application_code_configuration {
+      code_content {
+        text_content = "testCode\n"
+      }
+      code_content_type      = "PLAINTEXT"
+    }
+    sql_application_configuration {}
+  }
+
+  tags = {
+    firstTag  = "%s"
+    secondTag = "%s"
+  }
+}
+`, rInt, rInt, tag1, tag2),
+	)
+}
+
+func testAccKinesisAnalyticsV2ApplicationWithAddTags(rInt int, tag1, tag2, tag3 string) string {
+	return composeConfig(
+		testAccKinesisAnalyticsV2ApplicationConfigBase(rInt),
+		fmt.Sprintf(`
+data "aws_iam_policy_document" "assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["kinesisanalytics.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "kinesis_analyticsv2_application" {
+  name               = "tf-acc-test-%d-kinesis"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role_policy.json}"
+}
+
+resource "aws_kinesis_analyticsv2_application" "test" {
+  name                   = "testAcc-%d"
+  runtime                = "SQL-1_0"
+  service_execution_role = "${aws_iam_role.kinesis_analyticsv2_application.arn}"
+
+  application_configuration {
+    application_code_configuration {
+      code_content {
+        text_content = "testCode\n"
+      }
+      code_content_type      = "PLAINTEXT"
+    }
+    sql_application_configuration {}
+  }
+
+  tags = {
+    firstTag  = "%s"
+    secondTag = "%s"
+    thirdTag  = "%s"
+  }
+}
+`, rInt, rInt, tag1, tag2, tag3),
+	)
+}
+
+func testAccKinesisAnalyticsV2ApplicationWithConfigConflict(rInt int) string {
+	return composeConfig(
+		testAccKinesisAnalyticsV2ApplicationConfigBase(rInt),
+		fmt.Sprintf(`
+data "aws_iam_policy_document" "assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["kinesisanalytics.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "kinesis_analyticsv2_application" {
+  name               = "tf-acc-test-%d-kinesis"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role_policy.json}"
+}
+
+resource "aws_kinesis_analyticsv2_application" "test" {
+  name                   = "testAcc-%d"
+  runtime                = "SQL-1_0"
+  service_execution_role = "${aws_iam_role.kinesis_analyticsv2_application.arn}"
+
+  application_configuration {
+    application_code_configuration {
+      code_content {
+        text_content = "testCode\n"
+      }
+      code_content_type      = "PLAINTEXT"
+    }
+    flink_application_configuration {}
+    sql_application_configuration {}
+  }
+}
+`, rInt, rInt),
+	)
+}
+
+func testCheckResourceAttrUnknownSetId(resourceName, parentResource string, testResources []string, expectedVals []string) resource.TestCheckFunc {
```

Comment:
This function as well as `testCheckResourceAttrPairUnknownSetId` are kind of hacks around the difficulty of getting a nondeterministic set ID from a set. These set IDs are non-deterministic because they depend on nondeterministic resource ARNs.

I'm interested in whether or not there is a better way to get a set ID for cases like this.